packages/editor/src/ColorPicker.tsx

```diff
@@ -0,0 +1,214 @@
+  <label
+    {...props}
+    sx={{
+      display: 'block',
+      width,
+      fontSize: 10,
+      input: {
+        ':focus': {
+          color: 'primary',
+          bg: 'highlight',
+        },
+      },
+    }}
+  />
+)
+
+type CustomPickerForwardedProps = {
+  size?: number
+}
+type PickerProps = CustomPickerForwardedProps & CustomPickerProps
+
+export const Picker = CustomPicker<CustomPickerForwardedProps>(
+  ({ size = 256, ...props }) => {
+    return (
+      <div
+        sx={{
+          display: 'grid',
+          p: 2,
+          gap: 2,
+          width: size,
+          borderRadius: 4,
+          bg: 'white',
+          boxShadow: '0 2px 8px 1px rgba(0,0,0,.125)',
+        }}>
+        <div
+          sx={{
+            position: 'relative',
+            width: '100%',
+            height: 0,
+            paddingBottom: '75%',
+          }}>
+          <Saturation {...props} pointer={Lens} />
+        </div>
+        <div
+          sx={{
+            position: 'relative',
+            width: '100%',
+            height: 8,
+          }}>
+          <Hue {...props} pointer={Handle} />
+        </div>
+        <div
+          sx={{
+            display: 'flex',
+          }}>
+          <Label>
+            Hex
+            <Input
+              {...props}
+              value={props.hex}
+              name="hex"
+              label="hex"
+              onChange={val => {
+                props.onChange(val)
+              }}
+            />
+          </Label>
+          <Label>
+            Hue
+            <Input
+              {...props}
+              value={Math.round(props.hsl!.h)}
+              name="hue"
+              label="h"
+              // We need to cast to any because @types/react-color does not define types correctly
+              onChange={({ h }: any) => {
+                props.onChange({ ...props.hsl, h } as any)
+              }}
+            />
+          </Label>
+          <Label>
+            Saturation
+            <Input
+              {...props}
+              value={Math.round(props.hsl!.s * 100)}
+              name="saturation"
+              label="s"
+              // We need to cast to any because @types/react-color does not define types correctly
+              onChange={({ s }: any) => {
+                props.onChange({ ...props.hsl, s: s / 100 } as any)
+              }}
+            />
+          </Label>
+          <Label>
+            Lightness
+            <Input
+              {...props}
+              value={Math.round(props.hsl!.l * 100)}
+              name="lightness"
+              label="l"
+              // We need to cast to any because @types/react-color does not define types correctly
+              onChange={({ l }: any) => {
+                props.onChange({ ...props.hsl, l: l / 100 } as any)
+              }}
+            />
+          </Label>
+        </div>
+      </div>
+    )
+  }
+)
+
+type ColorPickerProps = React.PropsWithoutRef<PickerProps> & {
```

Comment:
Similarly as above (https://github.com/system-ui/theme-ui/pull/900/files#r426841405).

It's still visible to the user, and one can do `React.ComponentProps<typeof ColorPicker>`, but importing `ColorPickerProps` is just a bit nicer.

Reply:
Fixed in 14451fab.